platform/go/go-std/instrumented/src/handlers/handlers.go

```diff
@@ -0,0 +1,226 @@
+		if err := templates.FortuneTemplate.Execute(w, fortunes); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+	}
+}
+
+// FortuneHandlerPool . Test 4: Fortunes
+func FortuneHandlerPool(db storage.DB) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		fortunes, err := db.GetFortunesPool()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		fortunes = append(fortunes, templates.Fortune{Message: "Additional fortune added at request time."})
+
+		sort.Slice(fortunes, func(i, j int) bool {
+			return fortunes[i].Message < fortunes[j].Message
+		})
+
+		w.Header().Set("Server", "Go")
+		w.Header().Set("Content-Type", "text/html; charset=utf-8")
+		if err = templates.FortuneTemplate.Execute(w, fortunes); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		fortunes = fortunes[:0]
+		templates.FortunesPool.Put(fortunes)
+	}
+}
+
+// FortuneQuickHandler . Test 4: Fortunes
+func FortuneQuickHandler(db storage.DB) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		fortunes, err := db.GetFortunes()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		fortunes = append(fortunes, templates.Fortune{Message: "Additional fortune added at request time."})
+
+		sort.Slice(fortunes, func(i, j int) bool {
+			return fortunes[i].Message < fortunes[j].Message
+		})
+
+		w.Header().Set("Server", "Go")
+		w.Header().Set("Content-Type", "text/html; charset=utf-8")
+		templates.WriteFortunePage(w, fortunes)
+	}
+}
+
+// FortuneQuickHandlerPool . Test 4: Fortunes
+func FortuneQuickHandlerPool(db storage.DB) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		fortunes, err := db.GetFortunesPool()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		fortunes = append(fortunes, templates.Fortune{Message: "Additional fortune added at request time."})
+
+		sort.Slice(fortunes, func(i, j int) bool {
+			return fortunes[i].Message < fortunes[j].Message
+		})
+
+		w.Header().Set("Server", "Go")
+		w.Header().Set("Content-Type", "text/html; charset=utf-8")
+		templates.WriteFortunePage(w, fortunes)
+
+		fortunes = fortunes[:0]
+		templates.FortunesPool.Put(fortunes)
+	}
+}
+
+// UpdateHandler . Test 5: Database updates
+func UpdateHandler(db storage.DB) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		queries := queriesParam(r)
+		var err error
+
+		// worlds := make([]storage.World, queries)
+		worlds := storage.WorldsPool.Get().([]storage.World)[:queries]
+
+		// for _, world := range worlds {
+		for i := 0; i < queries; i++ {
+			func(i int) {
+				span := sentry.StartSpan(ctx, "db", func(s *sentry.Span) {
```

Comment:
Added bits of manual instrumentation in this file, and only this file and this HTTP handler, around DB operations, while we don't have direct SDK support for that.

Works to get us a list of child spans with every transaction. It's a pity we don't get to see a Sentry span waterfall chart in the report :)

Reply:
We could potentially grab https://preactjs.com/ and vendor in the sentry frontend code 👀

Reply:
Would that be doable without pulling in extra complexities from Sentry? Is the component well isolated? I am not familiar with the implementation.